stdlib/Distributed/test/distributed_exec.jl

```diff
@@ -132,6 +132,18 @@ end
 testf(id_me)
 testf(id_other)
 
+function loop_until_false(f::Function; timeout_seconds::Integer = 60)
+    start_time = time_ns()
+    while f() != false
```

Comment:
I feel some cognitive overhead looking at `!= false` or it's just me? :)

```suggestion
    while f()
```

Reply:
Can the function in question ever return `nothing`? Or is it guaranteed to return a `Bool`?

Reply:
We should enforce that it returns a bool.

Reply:
Now that we've renamed it to `poll_while`, I think it's clear that the caller should pass a function that returns a `Bool`.